src/File.php

```diff
@@ -132,6 +132,10 @@ class File extends DataObject implements AssetContainer, Thumbnail, CMSPreviewab
         "Owner" => Member::class,
     );
 
+    private static $indexes = array(
+        'FileHash' => true
```

Comment:
This can be a potentially expensive `dev/build` depending on how many files you have. I'm not too concerned, but wanted to point it out.

Reply:
Yes, that's a downside, but it speeds the other end when we search file by hash.

Reply:
FYI I confirm that MySQL will use the index for SQL expression like `HASH like 'ABCDEFGHIJ%'`.